### PR TITLE
website_quote: Various improvement in quotation template and ad…

### PR DESCRIPTION
**Pull request by rop-odoo**
Task: https://www.odoo.com/web?#id=23230&view_type=form&model=project.task&action=327&menu_id=4720
Pad: https://pad.odoo.com/p/r.51602d33434eabb6b6e17a0edd09b06e
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

…ded access rights to portal user for product.product and product.template model.

### CODE REVIEW

File: addons/website_quote/views/website_quotation.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
                     <tr>
                         <th>Products</th>
                         <th>Quantity</th>
-                        <th>Taxes</th>
+                        <th><span t-if="True in [len(line.tax_id) > 0 for line in quotation.order_line]">Taxes</span></th>
```
Comment (atp-odoo) [inline addons/website_quote/views/website_quotation.xml]: use "any" here. like
t-if="any([line.tax_id for line in quotation.order_line])"

File: addons/website_quote/views/website_quotation.xml [hunk 2]
```diff
@@ -544,10 +550,11 @@
   </template>
 
   <template id="optional_products">
-        <section t-if="quotation.options" id="options">
+      <t t-if="True in [(not option.line_id) for option in quotation.options]">
```
Comment (atp-odoo) [inline addons/website_quote/views/website_quotation.xml]: same as above. use any.

File: addons/product/security/ir.model.access.csv [hunk 3]
```diff
@@ -23,3 +23,5 @@ access_product_attribute_sale_manager,product.attribute manager,model_product_at
 access_product_attribute_value_sale_manager,product.attribute manager value,model_product_attribute_value,base.group_sale_manager,1,1,1,1
 access_product_attribute_price_sale_manager,product.attribute manager price,model_product_attribute_price,base.group_sale_manager,1,1,1,1
 access_product_attribute_line_sale_manager,product.attribute manager line,model_product_attribute_line,base.group_sale_manager,1,1,1,1
+access_product_product_portal,product.product.product,product.model_product_product,base.group_portal,1,0,0,0
```
Comment (tde-banana-odoo) [inline addons/product/security/ir.model.access.csv]: This does not read like a good idea to me.

File: addons/website_quote/views/website_quotation.xml [hunk 4]
```diff
@@ -64,9 +64,8 @@
                                         t-field-options='{"widget": "monetary", "display_currency": "quotation.pricelist_id.currency_id"}'
                                         t-att-style="line.discount and 'text-decoration: line-through' or ''"
                                         t-att-class="(line.discount and 'text-danger' or '') + 'text-right'"/>
-                                    <!-- TODO: apply monetary widget formating -->
                                     <div t-if="line.discount">
-                                        <t t-esc="'%.2f' % ((1-line.discount / 100.0) * line.price_unit)"/>
+                                        <t t-esc="float('%.2f' % ((1-line.discount / 100.0) * line.price_unit))" t-esc-options='{"widget": "monetary", "display_currency": "quotation.pricelist_id.currency_id"}'/>
```
Comment (tde-banana-odoo) [inline addons/website_quote/views/website_quotation.xml]: float ( string (float ) ) ?

File: addons/website_quote/views/website_quotation.xml [hunk 5]
```diff
@@ -555,10 +561,11 @@
   </template>
 
   <template id="optional_products">
-        <section t-if="quotation.options" id="options">
+      <t t-if="any([(not option.line_id) for option in quotation.options])">
+        <section id="options">
             <h1 class="page-header">Suggested Products</h1>
         </section>
-        <section t-if="quotation.options" id="options">
+        <section id="options">
```
Comment (tde-banana-odoo) [inline addons/website_quote/views/website_quotation.xml]: ID should be unique in an html document.

Comment (tde-banana-odoo) [inline addons/website_quote/views/website_quotation.xml]: Moreover, why adding an ID ?

File: addons/website_quote/views/website_quotation.xml [hunk 6]
```diff
@@ -590,7 +597,7 @@
                                     t-att-style="option.discount and 'text-decoration: line-through' or ''"
                                     t-att-class="option.discount and 'text-danger' or ''"/>
                                     <div t-if="option.discount">
-                                        <t t-esc="'%.2f' % ((1-option.discount / 100.0) * option.price_unit)"/>
+                                        <t t-esc="float('%.2f' % ((1-option.discount / 100.0) * option.price_unit))" t-esc-options='{"widget": "monetary", "display_currency": "quotation.pricelist_id.currency_id"}'/>
```
Comment (tde-banana-odoo) [inline addons/website_quote/views/website_quotation.xml]: Same remark as above.

**Pull request merged**